### ISSUE TITLE
[Bug]: Games Crashes on Load; MC version 1.20.1

### ISSUE THREAD

**Issue by windth001** [Toadstool-Studios/Creeper-Overhaul#114]
### Bug Description

The game crashed whilst rendering overlay
Error: net.minecraftforge.fml.ModLoadingException: Thermal Series (thermal) encountered an error during the done event phase

Suspected Mod: 
	Thermal Series (thermal), Version: 11.0.2

### Version

creeperoverhaul-3.0.1-forge

### Mod Loader Version

1.20.1 - 47.2.19

### Mod Loader

Forge

### Logs or additional context

Description: Rendering overlay

net.minecraftforge.fml.ModLoadingException: Thermal Series (thermal) encountered an error during the done event phase
§7java.lang.ClassCastException: class tech.thatgravyboat.creeperoverhaul.client.renderer.replaced.ReplacedCreeperRenderer cannot be cast to class net.minecraft.client.renderer.entity.LivingEntityRenderer (tech.thatgravyboat.creeperoverhaul.client.renderer.replaced.ReplacedCreeperRenderer is in module creeperoverhaul@3.0.1 of loader 'TRANSFORMER' @10745a02; net.minecraft.client.renderer.entity.LivingEntityRenderer is in module minecraft@1.20.1 of loader 'TRANSFORMER' @10745a02)
	at net.minecraftforge.fml.javafmlmod.FMLModContainer.acceptEvent(FMLModContainer.java:118) ~[javafmllanguage-1.20.1-47.2.19.jar%23440!/:?] {}
	at net.minecraftforge.fml.ModLoader.lambda$postEvent$29(ModLoader.java:326) ~[fmlcore-1.20.1-47.2.19.jar%23439!/:?] {}
	at java.lang.Iterable.forEach(Iterable.java:75) ~[?:?] {re:mixin}
	at net.minecraftforge.fml.ModList.forEachModInOrder(ModList.java:227) ~[fmlcore-1.20.1-47.2.19.jar%23439!/:?] {}
	at net.minecraftforge.fml.ModLoader.postEvent(ModLoader.java:326) ~[fmlcore-1.20.1-47.2.19.jar%23439!/:?] {}
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?] {}
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?] {}
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?] {}
	at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?] {}
	at net.optifine.reflect.Reflector.callVoid(Reflector.java:936) ~[OptiFine_1.20.1_HD_U_I6.jar%23496!/:?] {re:classloading}
	at net.optifine.reflect.ReflectorForge.postModLoaderEvent(ReflectorForge.java:373) ~[OptiFine_1.20.1_HD_U_I6.jar%23496!/:?] {re:classloading}
	at net.optifine.reflect.ReflectorForge.postModLoaderEvent(ReflectorForge.java:359) ~[OptiFine_1.20.1_HD_U_I6.jar%23496!/:?] {re:classloading}
	at net.minecraft.client.renderer.entity.EntityRenderDispatcher.m_6213_(EntityRenderDispatcher.java:506) ~[client-1.20.1-20230612.114412-srg.jar%23438!/:?] {re:mixin,pl:accesstransformer:B,xf:OptiFine:default,xf:fml:twilightforest:renderer,xf:fml:twilightforest:bake,re:classloading,pl:accesstransformer:B,xf:OptiFine:default,xf:fml:twilightforest:renderer,xf:fml:twilightforest:bake,pl:mixin:APP:playerAnimator-common.mixins.json:firstPerson.EntityRenderDispatcherMixin,pl:mixin:A}
	at net.minecraft.server.packs.resources.ResourceManagerReloadListener.m_10759_(ResourceManagerReloadListener.java:15) ~[client-1.20.1-20230612.114412-srg.jar%23438!/:?] {re:computing_frames,re:classloading,re:mixin}
	at java.util.concurrent.CompletableFuture$UniRun.tryFire(CompletableFuture.java:787) ~[?:?] {}
	at java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:482) ~[?:?] {}
	at net.minecraft.server.packs.resources.SimpleReloadInstance.m_143940_(SimpleReloadInstance.java:69) ~[client-1.20.1-20230612.114412-srg.jar%23438!/:?] {re:classloading}
	at net.minecraft.util.thread.BlockableEventLoop.m_6367_(BlockableEventLoop.java:198) ~[client-1.20.1-20230612.114412-srg.jar%23438!/:?] {re:mixin,pl:accesstransformer:B,xf:OptiFine:default,re:computing_frames,pl:accesstransformer:B,xf:OptiFine:default,re:classloading,pl:accesstransformer:B,xf:OptiFine:default}
	at net.minecraft.util.thread.ReentrantBlockableEventLoop.m_6367_(ReentrantBlockableEventLoop.java:23) ~[client-1.20.1-20230612.114412-srg.jar%23438!/:?] {re:mixin,re:computing_frames,re:classloading}
	at net.minecraft.util.thread.BlockableEventLoop.m_7245_(BlockableEventLoop.java:163) ~[client-1.20.1-20230612.114412-srg.jar%23438!/:?] {re:mixin,pl:accesstransformer:B,xf:OptiFine:default,re:computing_frames,pl:accesstransformer:B,xf:OptiFine:default,re:classloading,pl:accesstransformer:B,xf:OptiFine:default}
	at net.minecraft.util.thread.BlockableEventLoop.m_18699_(BlockableEventLoop.java:140) ~[client-1.20.1-20230612.114412-srg.jar%23438!/:?] {re:mixin,pl:accesstransformer:B,xf:OptiFine:default,re:computing_frames,pl:accesstransformer:B,xf:OptiFine:default,re:classloading,pl:accesstransformer:B,xf:OptiFine:default}
	at net.minecraft.client.Minecraft.m_91383_(Minecraft.java:1106) ~[client-1.20.1-20230612.114412-srg.jar%23438!/:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.client.Minecraft.m_91374_(Minecraft.java:718) ~[client-1.20.1-20230612.114412-srg.jar%23438!/:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.client.main.Main.main(Main.java:218) ~[1.20.1-forge-47.2.19.jar:?] {re:mixin,pl:runtimedistcleaner:A,re:classloading,pl:mixin:APP:flywheel.mixins.json:ClientMainMixin,pl:mixin:A,pl:runtimedistcleaner:A}
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?] {}
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?] {}
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?] {}
	at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?] {}
	at net.minecraftforge.fml.loading.targets.CommonLaunchHandler.runTarget(CommonLaunchHandler.java:111) ~[fmlloader-1.20.1-47.2.19.jar:?] {}
	at net.minecraftforge.fml.loading.targets.CommonLaunchHandler.clientService(CommonLaunchHandler.java:99) ~[fmlloader-1.20.1-47.2.19.jar:?] {}
	at net.minecraftforge.fml.loading.targets.CommonClientLaunchHandler.lambda$makeService$0(CommonClientLaunchHandler.java:25) ~[fmlloader-1.20.1-47.2.19.jar:?] {}
	at cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:30) ~[modlauncher-10.0.9.jar:?] {}
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:53) ~[modlauncher-10.0.9.jar:?] {}
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:71) ~[modlauncher-10.0.9.jar:?] {}
	at cpw.mods.modlauncher.Launcher.run(Launcher.java:108) ~[modlauncher-10.0.9.jar:?] {}
	at cpw.mods.modlauncher.Launcher.main(Launcher.java:78) ~[modlauncher-10.0.9.jar:?] {}
	at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:26) ~[modlauncher-10.0.9.jar:?] {}
	at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:23) ~[modlauncher-10.0.9.jar:?] {}
	at cpw.mods.bootstraplauncher.BootstrapLauncher.main(BootstrapLauncher.java:141) ~[bootstraplauncher-1.1.2.jar:?] {}
Caused by: java.lang.ClassCastException: class tech.thatgravyboat.creeperoverhaul.client.renderer.replaced.ReplacedCreeperRenderer cannot be cast to class net.minecraft.client.renderer.entity.LivingEntityRenderer (tech.thatgravyboat.creeperoverhaul.client.renderer.replaced.ReplacedCreeperRenderer is in module creeperoverhaul@3.0.1 of loader 'TRANSFORMER' @10745a02; net.minecraft.client.renderer.entity.LivingEntityRenderer is in module minecraft@1.20.1 of loader 'TRANSFORMER' @10745a02)
	at net.minecraftforge.client.event.EntityRenderersEvent$AddLayers.getRenderer(EntityRenderersEvent.java:194) ~[forge-1.20.1-47.2.19-universal.jar%23443!/:?] {re:classloading,pl:eventbus:A}
	at cofh.thermal.core.client.event.TCoreClientSetupEvents.addRenderLayers(TCoreClientSetupEvents.java:32) ~[thermal_core-1.20.1-11.0.2.18.jar%23492!/:11.0.2] {re:classloading}
	at cofh.thermal.core.client.event.__TCoreClientSetupEvents_addRenderLayers_AddLayers.invoke(.dynamic) ~[thermal_core-1.20.1-11.0.2.18.jar%23492!/:11.0.2] {re:classloading,pl:eventbus:B}
	at net.minecraftforge.eventbus.ASMEventHandler.invoke(ASMEventHandler.java:73) ~[eventbus-6.0.5.jar%2387!/:?] {}
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:315) ~[eventbus-6.0.5.jar%2387!/:?] {}
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:296) ~[eventbus-6.0.5.jar%2387!/:?] {}
	at net.minecraftforge.fml.javafmlmod.FMLModContainer.acceptEvent(FMLModContainer.java:114) ~[javafmllanguage-1.20.1-47.2.19.jar%23440!/:?] {}
	... 38 more

### Acknowledgement

- [X] I agree that I have read what was said above this check box.

**Comment by zagdrath** [comment]
Having the same exact issue currently, same mod conflicting.

**Comment by windth001** [comment]
> Having the same exact issue currently, same mod conflicting.

I went to the CoFH Team's Discord. People are reporting a conflict with Creeper Overhaul there as well. 
It was suggested to: Disable Creeper Overhaul's option to override the default creeper model or Disable the Thermal's vanilla holiday hats in the 'thermal-client' configs.

According to the persons who suggested this as a "fix" Creeper Overhaul has a client side-only functionality that is conflicting with the Thermal series creating an incompatibility.

"Creeper Overhaul replaces the creepers model with one that doesn't say its alive, and Thermal adds a hat to some mobs that say they are alive, but with the creepers model replaced the creeper says its alive but the model for the creeper says it isn't, so thermal gets confused"

Edit: Confirmed. Setting the holiday hats in the 'thermal-client' configs to false "fixed" the issue.
